src/rust/engine/src/intrinsics.rs

```diff
@@ -374,16 +374,15 @@ fn merge_digests_request_to_digest(
   let core = context.core;
   let store = core.store();
   async move {
-    let digests: Result<Vec<hashing::Digest>, String> = Python::with_gil(|py| {
-      let py_merge_digests = (*args[0]).as_ref(py);
-      externs::getattr::<Vec<&PyAny>>(py_merge_digests, "digests")
-        .unwrap()
-        .into_iter()
-        .map(|val| lift_directory_digest(val))
-        .collect()
-    });
+    let digests = Python::with_gil(|py| {
+      (*args[0])
+        .as_ref(py)
+        .extract::<PyRef<PyMergeDigests>>()
+        .map(|py_merge_digests| py_merge_digests.0.clone())
```

Comment:
Note this clone. It's because we want to move the `Vec<Digest>` out of the `Python::with_gil()` closure so that we can use it in an async call to `store.merge()` right after.

I think I'm confused what is _actually_ happening before. We before had a `&PyAny` and were extracting out `Vec<&PyAny>`, then iterating over and calling `.extract()` on each element into a `PyDigest`, unwrapped into a `Digest`. Then collecting into a `Vec<Digest`. 

I'm confused if that involved cloning each `PyDigest`. I'm pretty sure the answer is yes because this line extracts as a `PyDigest` rather than `PyRef<PyDigest>`:

https://github.com/pantsbuild/pants/blob/06efefc1006d15590bc3926ab0fc1d65e992d2fe/src/rust/engine/src/nodes.rs#L268-L271

I found in https://github.com/pantsbuild/pants/pull/13660 that calling `extract` on a `#[pyclass]` struct without wrapping in `PyRef<>` results in cloning it. And indeed, we derive `Clone` on `PyDigest`, so that all compiles.

So, I think this is no worse than before? Only more explicit.